microcosm_sagemaker/metrics/wandb/store.py

```diff
@@ -0,0 +1,46 @@
+from os import environ
+
+from microcosm_logging.decorators import logger
+
+from microcosm_sagemaker.decorators import metrics_observer, training_initializer
+from microcosm_sagemaker.hyperparameters import get_hyperparameters
+
+
+try:
+    import wandb
+except ImportError:
+    pass
+
+
+@training_initializer()
+@metrics_observer()
+@logger
+class WeightsAndBiases:
+    def __init__(self, graph):
+        self.graph = graph
+        self.testing = graph.metadata.testing
+        self.project_name = graph.metadata.name.replace("_", "-")
+
+    def init(self):
+        # Only initialize wandb if it is not a testing
+        if not self.testing:
+            # TODO: Remove this line if devops come up with a solution
+            # https://globality.atlassian.net/browse/DEVOPS-635
+            environ["WANDB_API_KEY"] = self.graph.config.wandb.api_key
```

Comment:
What is the advantage of setting the env var here vs just letting calling code set env var?  Is that because it's easier to put it in microcosm config sometimes?  Either way, might be slightly more usable to say
```suggestion
            if self.graph.config.wandb.api_key:
                environ["WANDB_API_KEY"] = self.graph.config.wandb.api_key
```
so that you don't overwrite it if they already put it there

Reply:
The calling code providers the API key, since it will be in the config of the ML models. But the _env var_ is set here in `microcosm-sagemaker`.